File: Laborator/Lab 4/Laborator 9/Laborator 9/Domain/Validators/StudentValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laborator_9.Domain.Validators
{
    class StudentValidator : IValidator<Student>
    {
        public void Validate(Student e)
        {
            string errors = "";
            if (e.ID < 0)
                errors += "Invalid ID ";
            if (e.Name == "")
                errors += "Invalid name ";
            if (e.School < 0)
                errors += "Invalid school ID ";

            if (errors.Length > 0)
            {
                Console.WriteLine(errors);
                throw new ValidationException(errors);
            }
        }
    }
}

File: Laborator/Lab 3/src/main/java/ubb/scs/map/ir/seminar/studentsmanagement/domain/validators/StudentValidator.java
package ubb.scs.map.ir.seminar.studentsmanagement.domain.validators;

import ubb.scs.map.ir.seminar.studentsmanagement.domain.Student;

public class StudentValidator implements Validator<Student> {
    @Override
    public void validate(Student entity) throws ValidationException {
        String errors = "";

        if (entity.getName().equals(""))
            errors += "Nume invalid ";

        if (entity.getMedia() < 1 || entity.getMedia() > 10)
            errors += "Media invalida";

        if (errors.length() > 0)
            throw new ValidationException(errors);
    }
}
